src/drivers/intel/cavs/alh.c

```diff
@@ -35,6 +35,19 @@ static int alh_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 	return 0;
 }
 
+/* get ALH hw params */
+static int alh_get_hw_params(struct dai *dai,
+			     struct sof_ipc_stream_params  *params)
+{
+	/* 0 means variable */
+	params->rate = 0;
```

Comment:
Btw, why is this variable on ALH and not SSP ? We may also want a macro for invalid and maybe set to -1 ?

Reply:
It is because during ```dai_set_config()``` for SSP we set specific value for ```ssp->params.fsync_rate``` as a result we cannot play streams with different rate. For ALH and HDA we do nothing during ```dai_set_config()```. IIUC from the firmware side, there is no information about how ALH or HDA was configured, so we assume that we can play anything (only need is to configure dma in the same way on the firmware and host size).